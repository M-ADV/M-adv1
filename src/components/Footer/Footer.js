import React from "react";
import { Link } from "react-router-dom";

export default function Footer() {
  return (
    <>
      <div>
        <footer
          className="text-light text-center text-lg-start Footer "
          style={{ backgroundColor: "#870000" }}
        >
          <div className="container p-4">
            <div className="row mt-4">
              <div className="col-lg-4 col-md-12 mb-4 mb-md-0">
                <h5 className="text-uppercase mb-4">
                  <b>About Company</b>
                </h5>

                <p>
                  Welcome to M-Advisor, your trusted source for mortgage loans.
                  We're committed to helping you achieve your dream of
                  homeownership by providing personalized, expert support and a
                  wide range of loan options.
                </p>

                <div className="mt-4">
                  <i className="bi bi-facebook fs-2"></i>

                  <i className="bi bi-youtube fs-2 mx-3"></i>

                  <i className="bi bi-twitter   fs-2 mx-3"></i>

                  <i className="bi bi-whatsapp  fs-2 mx-3"></i>
                </div>
              </div>

              <div className="col-lg-4 col-md-6 mb-4 mb-md-0">
                <h5 className="text-uppercase mb-4 pb-1">
                  <b>Search something</b>
                </h5>

                <div className="input-group text-dark mb-4">
                  <input
                    type="text"
                    id="formControlLg"
                    className="form-control form-control-lg"
                    placeholder="Search"
                  />
                </div>

                <ul className="fa-ul" style={{ marginLeft: "1.65em" }}>
                  <li className="mb-3">
                    <span className="fa-li">
                      <i className="bi bi-house  fs-3 "></i>
                    </span>
                    <span className="ms-3">Adibatla,Hyderabad,501510</span>
                  </li>
                  <li className="mb-3">
                    <span className="fa-li">
                      <i className="bi bi-envelope fs-3"></i>
                    </span>
                    <span className="ms-3">MAdvisor@outlook.com</span>
                  </li>
                  <li className="mb-3">
                    <span className="fa-li">
                      <i className="bi bi-telephone-inbound fs-3 "></i>
                    </span>
                    <span className="ms-3 my-auto ">+919912622103 </span>
                  </li>
                </ul>
              </div>

              <div className="col-lg-4 col-md-6 mb-4 mb-md-0">
                <h5 className="text-uppercase mb-4">
                  <b>Opening hours</b>
                </h5>

                <table className="table text-center text-white">
                  <tbody className="fw-normal">
                    <tr>
                      <td>Mon - Fri:</td>
                      <td>8am - 9pm</td>
                    </tr>
                    <tr>
                      <td>Sat - Sun:</td>
                      <td>closed</td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>

          <div
            className="text-center p-3 footer"
            // style={{ backgroundColor: "rgba(0, 0, 0, 0.2)" }}
          >
            &#169; 2023 Copyright:
            <Link className="text-danger">M-Loan.com</Link>
          </div>
        </footer>
      </div>
    </>
  );
}
