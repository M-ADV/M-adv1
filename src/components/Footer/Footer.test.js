import { getAllByRole, render, screen } from "@testing-library/react";
import { renderComponent } from "../../utils/test_helpers";
import Footer from "./Footer";

describe("Footer Page", () => {
  test("should render Intial page", () => {
    render(renderComponent(<Footer />));
    const text = screen.getAllByText(
      /Welcome to M-Loan, your trusted source for mortgage loans./i
    );
    expect(text[0]).toBeInTheDocument();

    expect(
      screen.getByRole("heading", { name: /Search something/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: /About Company/i })
    ).toBeInTheDocument();
  });
  test("should render intial page roles", () => {
    render(renderComponent(<Footer />));
    expect(
      screen.getByRole("link", { name: "M-Loan.com" })
    ).toBeInTheDocument();

    expect(screen.getByPlaceholderText("Search")).toBeInTheDocument();
  });
  test("should render correct business hours", () => {
    render(renderComponent(<Footer />));

    expect(screen.getByText("Mon - Fri:")).toBeInTheDocument();
  });
  test("should test snapshot", () => {
    //render( renderComponent(<Footer/>));
    const { asFragment } = render(renderComponent(<Footer />));
    expect(asFragment()).toMatchSnapshot();
  });
});
