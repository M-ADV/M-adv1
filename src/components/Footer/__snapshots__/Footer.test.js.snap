// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Page should test snapshot 1`] = `
<DocumentFragment>
  <div>
    <footer
      class="text-light text-center text-lg-start Footer "
      style="background-color: rgb(135, 0, 0);"
    >
      <div
        class="container p-4"
      >
        <div
          class="row mt-4"
        >
          <div
            class="col-lg-4 col-md-12 mb-4 mb-md-0"
          >
            <h5
              class="text-uppercase mb-4"
            >
              <b>
                About Company
              </b>
            </h5>
            <p>
              Welcome to M-Advisor, your trusted source for mortgage loans. We're committed to helping you achieve your dream of homeownership by providing personalized, expert support and a wide range of loan options.
            </p>
            <div
              class="mt-4"
            >
              <i
                class="bi bi-facebook fs-2"
              />
              <i
                class="bi bi-youtube fs-2 mx-3"
              />
              <i
                class="bi bi-twitter   fs-2 mx-3"
              />
              <i
                class="bi bi-whatsapp  fs-2 mx-3"
              />
            </div>
          </div>
          <div
            class="col-lg-4 col-md-6 mb-4 mb-md-0"
          >
            <h5
              class="text-uppercase mb-4 pb-1"
            >
              <b>
                Search something
              </b>
            </h5>
            <div
              class="input-group text-dark mb-4"
            >
              <input
                class="form-control form-control-lg"
                id="formControlLg"
                placeholder="Search"
                type="text"
              />
            </div>
            <ul
              class="fa-ul"
              style="margin-left: 1.65em;"
            >
              <li
                class="mb-3"
              >
                <span
                  class="fa-li"
                >
                  <i
                    class="bi bi-house  fs-3 "
                  />
                </span>
                <span
                  class="ms-3"
                >
                  Adibatla,Hyderabad,501510
                </span>
              </li>
              <li
                class="mb-3"
              >
                <span
                  class="fa-li"
                >
                  <i
                    class="bi bi-envelope fs-3"
                  />
                </span>
                <span
                  class="ms-3"
                >
                  MAdvisor@outlook.com
                </span>
              </li>
              <li
                class="mb-3"
              >
                <span
                  class="fa-li"
                >
                  <i
                    class="bi bi-telephone-inbound fs-3 "
                  />
                </span>
                <span
                  class="ms-3 my-auto "
                >
                  +919912622103 
                </span>
              </li>
            </ul>
          </div>
          <div
            class="col-lg-4 col-md-6 mb-4 mb-md-0"
          >
            <h5
              class="text-uppercase mb-4"
            >
              <b>
                Opening hours
              </b>
            </h5>
            <table
              class="table text-center text-white"
            >
              <tbody
                class="fw-normal"
              >
                <tr>
                  <td>
                    Mon - Fri:
                  </td>
                  <td>
                    8am - 9pm
                  </td>
                </tr>
                <tr>
                  <td>
                    Sat - Sun:
                  </td>
                  <td>
                    closed
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        class="text-center p-3 footer"
      >
        Â© 2023 Copyright:
        <a
          class="text-danger"
          href="/"
        >
          M-Loan.com
        </a>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
